examples/model_selection/plot_multiple_metric_evaluation.py

```diff
@@ -0,0 +1,97 @@
+"""Demonstration of multimetric evaluation on cross_val_score and GridSearchCV
+
+Multiple metric grid search (or random search) can be done by setting the
+``scoring`` parameter to a list of metric scorer names or a dict mapping the
+scorer names to the scorer callables.
+
+The scores of all the scorers are available in the ``cv_results_`` dict at keys
+ending in ``'_<scorer_name/metric_name>'`` (``'mean_test_precision'``,
+``'rank_test_precision'``, etc...)
+
+The ``best_estimator_`` and ``best_index_`` attributes are now
+a dict mapping the scorer name to the best estimator and the best candidate
+index corresponding to the best score for that metric which is stored in
+``best_score_[<scorer_name/metric_name>]``.
+
+Similarly the :func:`sklearn.model_selection.cross_val_score`,
+:func:`sklearn.model_selection.learning_curve` and
+:func:`sklearn.model_selection.validation_curve` all support multiple metric
+evaluation when the scoring parameter is a list/tuple of strings denoting
+predefined scorers or a dict mapping the scorer names to the scorer callables.
+"""
+
+# Author: Raghav RV <rvraghav93@gmail.com>
+# License: BSD
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+from sklearn.datasets import make_hastie_10_2
+from sklearn.model_selection import GridSearchCV
+from sklearn.metrics import precision_score
+from sklearn.metrics.scorer import make_scorer
+from sklearn.tree import DecisionTreeClassifier
+
+
+print(__doc__)
+
+
+X, y = make_hastie_10_2(n_samples=8000, random_state=42)
+
+# The scorers can be a std scorer referenced by its name or one wrapped
+# by sklearn.metrics.scorer.make_scorer
+scoring = {'AUC Score': 'roc_auc', 'Precision': make_scorer(precision_score),
+           'recall': 'recall', 'F1 Score': 'f1'}
```

Comment:
I intentionally kept it this way to demonstrate that keys can be the same as scorer names. Do you think it's better to capitalize still?